proposals/utils/stepper_helpers.py

```diff
@@ -251,24 +282,38 @@ def instantiate_form(self):
         )
         return model_form
 
-    def get_errors(self):
+    def get_errors(self, include_children=False):
         """
-        By default, this returns the errors of the instantiated form, which get
-        evaluated as a boolean. By passing a custom get_checker_errors function
-        from the checker, custom errors can be specified also.
+        By default, this returns the errors of the instantiated form and,
+        optionally, its children. This then gets evaluated as a boolean.
+        By passing a custom get_checker_errors function
+        from the checker, custom errors also be added.
         """
-        if hasattr(self, "get_checker_errors"):
-            return self.get_checker_errors() or self.model_form.errors
-        return self.model_form.errors
+        errors = []
+        errors += self.model_form.errors
 
-    def css_classes(self):
-        css_classes = super().css_classes()
+        if hasattr(self, "get_checker_errors"):
+            errors += self.get_checker_errors()
+        if include_children and self.children:
+            for child in self.children:
+                errors += child.get_errors(include_children=True)
+        return errors
 
-        if self.get_errors():
-            css_classes += " incomplete"
+    def build_css_classes(self):
+        """
+        If an item is collapsed, its completeness also depends on
+        its children, otherwise, it only depends on the form the
+        the item represents.
+        """
+        if not self.is_expanded and self.children:
+            if self.get_errors(include_children=True):
+                self.css_classes.add("incomplete")
+            else:
+                self.css_classes.add("complete")
+        elif self.get_errors():
+            self.css_classes.add("incomplete")
         else:
-            css_classes += " complete"
-        return css_classes
+            self.css_classes.add("complete")
```

Comment:
I also feel like this could be simplified now that we have the `is_expanded` variable.

Wouldn't something like `if self.get_errors(include_children=self.is_expanded)` remove the need for the double `if` statements?

I might be missing something though, just a quick suggestion.